src/storm/mesos/MesosNimbus.java

```diff
@@ -97,57 +105,93 @@ public String getHostName(Map<String, SupervisorDetails> map, String nodeId) {
     return nodeId;
   }
 
+  @SuppressWarnings("unchecked")
   @Override
   public void prepare(Map conf, String localDir) {
     try {
-      _conf = conf;
-      _state = new LocalState(localDir);
-      String id = (String) _state.get(FRAMEWORK_ID);
-
-      _allowedHosts = listIntoSet((List) _conf.get(CONF_MESOS_ALLOWED_HOSTS));
-      _disallowedHosts = listIntoSet((List) _conf.get(CONF_MESOS_DISALLOWED_HOSTS));
-
-      _scheduler = new NimbusScheduler();
-
-      Number failoverTimeout = Optional.fromNullable((Number) conf.get(CONF_MASTER_FAILOVER_TIMEOUT_SECS)).or(3600);
-      String role = Optional.fromNullable((String) conf.get(CONF_MESOS_ROLE)).or("*");
-      Boolean checkpoint = Optional.fromNullable((Boolean) conf.get(CONF_MESOS_CHECKPOINT)).or(false);
-      String framework_name = Optional.fromNullable((String) conf.get(CONF_MESOS_FRAMEWORK_NAME)).or("Storm!!!");
-
-      FrameworkInfo.Builder finfo = FrameworkInfo.newBuilder()
-          .setName(framework_name)
-          .setFailoverTimeout(failoverTimeout.doubleValue())
-          .setUser("")
-          .setRole(role)
-          .setCheckpoint(checkpoint);
-
-      if (id != null) {
-        finfo.setId(FrameworkID.newBuilder().setValue(id).build());
-      }
-
-      Integer port = (Integer) _conf.get(CONF_MESOS_LOCAL_FILE_SERVER_PORT);
-      LOG.info("Using local port: " + port);
-      _localFileServerPort = Optional.fromNullable(port);
-
-      _httpServer = new LocalFileServer();
-      _configUrl = _httpServer.serveDir("/conf", "conf", _localFileServerPort);
-      LOG.info("Started serving config dir under " + _configUrl);
-
-      MesosSchedulerDriver driver =
-          new MesosSchedulerDriver(
-              _scheduler,
-              finfo.build(),
-              (String) conf.get(CONF_MASTER_URL));
-
-      driver.start();
-      LOG.info("Waiting for scheduler to initialize...");
-      _scheduler.waitUntilRegistered();
-      LOG.info("Scheduler initialized...");
+        initialize(conf,localDir);
+
+        MesosSchedulerDriver driver = createMesosDriver();
+        
+        driver.start();
+      
+        LOG.info("Waiting for scheduler to initialize...");
+        
+        getScheduler().waitUntilRegistered();
+        
+        LOG.info("Scheduler initialized...");
+      
     } catch (Exception e) {
       throw new RuntimeException(e);
     }
   }
-
+  
+  @SuppressWarnings("unchecked")
+  protected void initialize(Map conf, String localDir) throws Exception {
+      
+      setConfig(conf);
+      setLocalState(new LocalState(localDir));
+
+      _allowedHosts = listIntoSet((List<String>)conf.get(CONF_MESOS_ALLOWED_HOSTS));
```

Comment:
>>> I wonder if listIntoSet can be replaced with google's gauve
Sets.newHashSet?

Here is a place where I would use a setter/getter. If you use

setAllowedHost (List<String> list) {

      if (list == null)

          _allowedHost = new HashSet<String>(0);

     else

         _allowedHost = new HashSet(list);

}


then in (Assume the same for disallowedHost)


public boolean isHostAccepted(String hostname) {
    return
        (_allowedHosts == null && _disallowedHosts == null) ||
            (_allowedHosts != null && _allowedHosts.contains(hostname)) ||
            (_disallowedHosts != null &&
!_disallowedHosts.contains(hostname))
        ;
  }


it would be

public boolean isHostAccepted(String hostname) {
    return (getAllowedHost().contains(hostname) ||
!getDisallowedHost().contains(hostname);
  }


The contract would hold up as long as the getters/setters were used.


Thoughts?



On Tue, May 5, 2015 at 11:59 AM, Paul Read <pdread101@gmail.com> wrote:

> In src/storm/mesos/MesosNimbus.java
> <https://github.com/mesos/storm/pull/37#discussion_r29535084>:
>
> >      } catch (Exception e) {
> >        throw new RuntimeException(e);
> >      }
> >    }
> > -
> > +
> > +  @SuppressWarnings("unchecked")
> > +  protected void initialize(Map conf, String localDir) throws Exception {
> > +
> > +      setConfig(conf);
> > +      setLocalState(new LocalState(localDir));
> > +
> > +      _allowedHosts = listIntoSet((List<String>)conf.get(CONF_MESOS_ALLOWED_HOSTS));
>
> >>> I wonder if listIntoSet can be replaced with google's gauve
> Sets.newHashSet?
>
>
> No, Sets.newHashSet throws a NullPointerException if the list is null.
>
> Thanks
>
>
>
>
> On Fri, May 1, 2015 at 5:28 PM, Timothy Chen <notifications@github.com>
> wrote:
>
>> In src/storm/mesos/MesosNimbus.java
>> <https://github.com/mesos/storm/pull/37#discussion_r29535084>:
>>
>> >      } catch (Exception e) {
>> >        throw new RuntimeException(e);
>> >      }
>> >    }
>> > -
>> > +
>> > +  @SuppressWarnings("unchecked")
>> > +  protected void initialize(Map conf, String localDir) throws Exception {
>> > +
>> > +      setConfig(conf);
>> > +      setLocalState(new LocalState(localDir));
>> > +
>> > +      _allowedHosts = listIntoSet((List<String>)conf.get(CONF_MESOS_ALLOWED_HOSTS));
>>
>> I wonder if listIntoSet can be replaced with google's gauve
>> Sets.newHashSet?
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/mesos/storm/pull/37/files#r29535084>.
>>
>
>